typed/merge_test.go

```diff
@@ -378,3 +386,57 @@ func TestMerge(t *testing.T) {
 		})
 	}
 }
+
+func TestMergeUninitialized(t *testing.T) {
+	parser, err := typed.NewParser(typed.YAMLObject(`types:
+- name: stringPair
+  struct:
+    fields:
+    - name: key
+      type:
+        scalar: string
+    - name: value
+      type:
+        untyped: {}
+`))
+	if err != nil {
+		t.Fatalf("Failed to create parser: %v", err)
+	}
+	uninit := typed.TypedValue{}
+	init, err := parser.FromYAML(typed.YAMLObject(`{"key":"foo","value":{}}`), "stringPair")
+	if err != nil {
+		t.Fatalf("Failed to parse yaml object: %v", err)
+	}
+
+	got, err := uninit.Merge(init)
+	if err != nil {
+		t.Fatalf("got validation errors: %v", err)
+	}
+	if !reflect.DeepEqual(got.AsValue().ToUnstructured(true), init.AsValue().ToUnstructured(true)) {
+		t.Errorf("Expected\n%v\nbut got\n%v\n",
+			init.AsValue().HumanReadable(), got.AsValue().HumanReadable(),
+		)
+	}
+
+	// Change the order, uninit is init, init is uninit, same result.
+	got, err = init.Merge(uninit)
```

Comment:
If you test Compare instead of Merge it'll show the thing I'm complaining about :)

Reply:
Like the Comparison.IsSame would return true...